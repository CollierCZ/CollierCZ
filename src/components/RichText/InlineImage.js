import PropTypes from 'prop-types'
import React from 'react'

const InlineImage = ({ description, url }) => {
  return (
    <picture>
    <source srcSet={url}
            media="(min-width: 1200px)" />
            <source srcSet={url+'?w=720&fit=scale'}
                    media="(min-width: 900px)" />
        <source srcSet={url+'?w=500&fit=scale'}
                media="(min-width: 500px)" />
        <img
            src={url+'?w=320&fit=scale'}
            alt={description}
        />
    </picture>
  )
}

InlineImage.propTypes = {
  description: PropTypes.string,
  url: PropTypes.string.isRequired
}

export default InlineImage