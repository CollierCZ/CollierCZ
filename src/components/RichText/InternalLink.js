import { Link } from 'gatsby'
import PropTypes from 'prop-types'
import React from 'react'

const InternalLink = ({ content, linkId, type, urlSlug }) => {
  let url;

  switch (type) {
    case 'article': {
      url = `/articles/${urlSlug}`;
      break;
    }

    default: {
      url = '/not-found';
    }
  }

  return (
    <Link key={linkId} to={url}>
      {content}
    </Link>
  );
}

InternalLink.propTypes = {
  content: PropTypes.string.isRequired,
  linkId: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  urlSlug: PropTypes.string.isRequired
}

export default InternalLink