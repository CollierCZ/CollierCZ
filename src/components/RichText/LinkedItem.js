import { css } from "@emotion/core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faQuoteLeft, faQuoteRight } from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import React from "react";

const LinkedItem = ({ linkedItem }) => {
  const type = linkedItem.system.type;

  switch (type) {
    case "quote": {
      const source = linkedItem.elements.source.resolvedHtml
        ? `<footer>${linkedItem.elements.source.resolvedHtml}</footer>`
        : "";
      const quote = source
        ? linkedItem.elements.quote.resolvedHtml + source
        : linkedItem.elements.quote.resolvedHtml;
      return (
        <div
          css={css`
            display: flex;
            svg {
              font-size: 5rem;
            }
            footer {
              font-size: 2rem;
            }
          `}
        >
          <FontAwesomeIcon icon={faQuoteLeft} />
          <blockquote
            dangerouslySetInnerHTML={{ __html: quote }}
            css={css`
              font-style: normal;
              margin: 0 3rem;
            `}
          />
          <FontAwesomeIcon icon={faQuoteRight} />
        </div>
      );
    }

    default:
      return null;
  }
};

LinkedItem.propTypes = {
  linkedItem: PropTypes.object.isRequired
};

export default LinkedItem;
