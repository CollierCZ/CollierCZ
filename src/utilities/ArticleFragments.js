import { graphql } from "gatsby"

export const ArticleListFragment = graphql`
  fragment ArticleListFragment on KontentItemArticle {
    fields {
      slug
      tags
      date
    }
    elements {
      metadata__description {
        value
      }
      metadata__keywords {
        value
      }
      categories {
        value {
          name
        }
      }
      title {
        value
      }
      teaser {
        value {
          url
        }
      }            
    }
  }
`