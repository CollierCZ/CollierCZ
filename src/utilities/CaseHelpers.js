const slugify = (text) => {
  return text
    .toString()
    .toLowerCase()
    .trim()
    .replace(/\s+/g, '-')       // Replace spaces with -
    .replace(/[^\w-]+/g, '')     // Remove all non-word chars
}

const titleCase = (text) => {
  return(
    text.replace(/\w\S*/g, function(txt) {
      return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase();
    })
  )
}

module.exports =
{
  slugify: slugify,
  titleCase: titleCase
}