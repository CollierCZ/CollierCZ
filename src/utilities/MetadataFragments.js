import { graphql } from "gatsby"

export const MetadataFragment = graphql `
  fragment MetadataFragment on KontentItemHome {
    elements {
      title {
        value
      }
      splash_image {
        value {
          url
        }
      }
      blog_logo {
        value {
          url
        }
      }
      metadata__description {
        value
      }
      socialmedia {
        value
      }
      base_url {
        value
      }
    }
  }
`

export const SocialMediaFragment = graphql `
  fragment SocialMediaFragment on KontentItemHome {
    elements {
      socialmedia {
        value
      }
    }
  }
`