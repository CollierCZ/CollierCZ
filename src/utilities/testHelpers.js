import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { navigate } from "gatsby";

export const simpleRenderer = component => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
};
export const clickTextLink = component => {
  const mountedComponent = mount(component);
  mountedComponent
    .find("a")
    .first()
    .simulate("click");
  expect(navigate).toHaveBeenCalled();
};
