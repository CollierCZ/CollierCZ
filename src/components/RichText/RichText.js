import parseHTML, { domToReact } from "html-react-parser";
import InlineImage from "./InlineImage";
import LinkedItem from "./LinkedItem";
import { navigate } from "gatsby";
import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import { defaultTokens } from "@kiwicom/orbit-design-tokens"
import Heading from "@kiwicom/orbit-components/lib/Heading";
import List from "@kiwicom/orbit-components/lib/List";
import ListItem from "@kiwicom/orbit-components/lib/List/ListItem";
import Text from "@kiwicom/orbit-components/lib/Text";
import TextLink from "@kiwicom/orbit-components/lib/TextLink";

export const StyledInlineCode = styled.code`
  background-color: ${defaultTokens.paletteCloudNormal};
  border-radius: ${defaultTokens.borderRadiusNormal};
  font-family: "Roboto Mono", "Liberation Mono", monospace;
  padding: ${defaultTokens.spaceXXXSmall} ${defaultTokens.spaceXXSmall};
`

const RichText = ({ content, images, links, linkedItems, stopPropagation }) => {
  if (!content || !content.length) {
    return null;
  }
  // Parse HTML as React components, replacing any content items.
  const children = parseHTML(content, {
    replace: (domNode) =>
      replaceNode(domNode, images, links, linkedItems, stopPropagation),
  });

  return <>{children}</>;
};

const replaceLists = {
  replace: ({ attribs, children, name }) => {
    if (name === "ul") {
      return (
        <List spaceAfter={"small"}>{domToReact(children, replaceLists)}</List>
      );
    }
    if (name === "li") {
      const content = children || null;
      return (
        <ListItem spaceAfter={"smallest"}>
          {domToReact(content, replaceLists)}
        </ListItem>
      );
    }

    if (name === "a") {
      return replaceLinks(attribs, children);
    }
  },
};

const replaceLinks = (attribs, children, links, stopPropagation) => {
  // Replace internal links
  if (attribs["data-item-id"]) {
    const id = attribs["data-item-id"] || null;
    const link = links.find((link) => link.link_id === id);
    const url = `/articles/${link.url_slug}`;
    return (
      <TextLink
        href={url}
        onClick={(event) => {
          event.preventDefault();
          navigate(url);
        }}
        key={url}
        stopPropagation={stopPropagation}
      >
        {children[0].data}
      </TextLink>
    );
  }

  //Replace external links
  const url = attribs["href"] || null;
  return (
    <TextLink href={url} key={url} stopPropagation={stopPropagation}>
      {children[0].data}
    </TextLink>
  );
};

RichText.propTypes = {
  content: PropTypes.string.isRequired,
  images: PropTypes.arrayOf(PropTypes.object),
  links: PropTypes.arrayOf(PropTypes.object),
  linkedItems: PropTypes.arrayOf(PropTypes.object),
  stopPropagation: PropTypes.bool,
};

export default RichText;

/** Replace HTML DOM node with React component. */
function replaceNode(domNode, images, links, linkedItems, stopPropagation) {
  const { attribs, children, name } = domNode;
  let content = [];
  // Replace inline assets.
  if (name === "figure") {
    const id = attribs["data-asset-id"] || null;
    const image = images.find((image) => image.image_id === id);
    return <InlineImage description={image.description} url={image.url} />;
  }

  // Replace inline linked items and components.
  if (attribs && attribs["data-type"] === "item") {
    const codename = attribs["data-codename"] || null;
    const linkedItem = linkedItems.find(
      (item) => item.system.codename === codename
    );
    return <LinkedItem linkedItem={linkedItem} />;
  }

  // Replace paragraphs.
  if (domNode.name === "p") {
    children.forEach((node, index) => {
      if (node.name === "a") {
        content.push(
          replaceLinks(node.attribs, node.children, links, stopPropagation)
        );
      }
      if (node.type === "text") {
        content.push(node.data);
      }
      if (node.name === "code" && node.type === "tag") {
        content.push(<StyledInlineCode key={index}>{node.children[0].data}</StyledInlineCode>)
      }
      return "hi";
    });
    return <Text spaceAfter={"small"}>{content}</Text>;
  }

  // Replace headings.
  const headings = ["h2", "h3", "h4", "h5", "h6"];
  if (headings.indexOf(name) > -1) {
    return (
      <Heading
        element={name}
        type={"title" + name.slice(1)}
        spaceAfter={"small"}
      >
        {domToReact(children)}
      </Heading>
    );
  }

  // Recursively replace unordered lists.
  if (name === "ul") {
    return (
      <List spaceAfter={"small"}>{domToReact(children, replaceLists)}</List>
    );
  }
}
