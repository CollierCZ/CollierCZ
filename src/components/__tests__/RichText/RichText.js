import React from "react";
import RichText from "../../RichText/RichText";
import { clickTextLink, simpleRenderer } from "../../../utilities/testHelpers";

const textWithLinks = {
  content:
    '<p>Read a <a data-item-id="3cc9d486-53e2-495d-9e2b-25665e0ff29f" href="">text version of the talk</a> I gave at the <a href="http://www.writethedocs.org/conf/prague/2019/">Write the Docs Prague 2019</a> conference.</p>',
  images: [],
  links: [
    {
      linkId: "3cc9d486-53e2-495d-9e2b-25665e0ff29f",
      urlSlug: "seeing-your-docs-through-different-eyes",
      type: "article",
      codename: "write_the_docs_2019"
    }
  ],
  linkedItems: []
};

const textWithHeading =
  "<h2>An awesome heading</h2><p>A video of a talk I gave</p>";

const textWithImage = {
  content:
    '<p>An awesome intro</p><figure data-asset-id="d11df065-aadc-4bb4-881d-c0ee5f13d2ad" data-image-id="d11df065-aadc-4bb4-881d-c0ee5f13d2ad"><img src="https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/18b254b8-ee42-42ca-9c5a-9105fa9a473d/Slide6.PNG" data-asset-id="d11df065-aadc-4bb4-881d-c0ee5f13d2ad" data-image-id="d11df065-aadc-4bb4-881d-c0ee5f13d2ad" alt="A map of an experience of a plane trip from London To New York."></figure>',
  images: [
    {
      description:
        "A map of an experience of a plane trip from London To New York.",
      url:
        "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/18b254b8-ee42-42ca-9c5a-9105fa9a473d/Slide6.PNG",
      imageId: "d11df065-aadc-4bb4-881d-c0ee5f13d2ad"
    }
  ]
};

const textWithLinkedItem = {
  content:
    '<p>An intro to a component.</p>\n<p type="application/kenticocloud" data-type="item" data-rel="component" data-codename="d03cd933_61c3_01aa_c928_be6bcd158c5a" data-sdk-resolved="1" class="kc-linked-item-wrapper"></p>',
  images: [],
  links: [],
  linkedItems: [
    {
      id: "0afa8bf7-ba80-5a4d-8b31-597de005dab4",
      system: {
        codename: "d03cd933_61c3_01aa_c928_be6bcd158c5a",
        type: "quote"
      },
      elements: {
        quote: {
          resolvedData: {
            html:
              "<p>You never really understand a person until you consider things from [their] point of view â€” until you climb into [their] skin and walk around in it.</p>"
          }
        },
        source: {
          resolvedData: {
            html:
              "<p>Atticus Finch in <em>To Kill a Mockingbird </em>by Harper Lee</p>"
          }
        }
      }
    }
  ]
};

const textWithList = {
  content:
    "<ul><li>A video of a talk I gave at the Write the Docs Prague 2019 conference.</li><li> Or read a text version of the talk.</li></ul>",
  images: [],
  links: [],
  linkedItems: []
};

const textWithListWithLinks = {
  content:
    '<ul><li>A video of a talk I gave at the <a href="http://www.writethedocs.org/conf/prague/2019/">Write the Docs Prague 2019</a> conference.</li><li> Or read a text version of the talk.</li></ul>',
  images: [],
  links: [],
  linkedItems: []
};

describe("RichText", () => {
  it("correctly returns null on empty", () => {
    simpleRenderer(<RichText content={''}/>);
  });
  it("renders correctly with links", () => {
    simpleRenderer(<RichText {...textWithLinks} />);
  });
  it("renders a working internal link correctly", () => {
    clickTextLink(<RichText {...textWithLinks} />);
  });
  it("renders correctly with a heading", () => {
    simpleRenderer(<RichText content={textWithHeading} />);
  });
  it("renders correctly with an image", () => {
    simpleRenderer(<RichText {...textWithImage} />);
  });
  it("renders correctly with a linked item", () => {
    simpleRenderer(<RichText {...textWithLinkedItem} />);
  });
  it("renders correctly with a simple list", () => {
    simpleRenderer(<RichText {...textWithList} />);
  });
  it("renders correctly with a list with links", () => {
    simpleRenderer(<RichText {...textWithListWithLinks} />);
  });
});
