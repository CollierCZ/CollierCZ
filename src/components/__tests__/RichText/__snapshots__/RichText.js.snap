// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichText correctly returns null on empty 1`] = `null`;

exports[`RichText renders correctly with a heading 1`] = `
Array [
  .c0 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 22px;
  font-weight: 500;
  color: #252A31;
  line-height: 28px;
  margin: 0;
  margin-bottom: 8px;
}

<div
    className="c0"
  >
    An awesome heading
  </div>,
  .c0 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #252A31;
  line-height: 20px;
  text-align: left;
  margin: 0;
  margin-bottom: 8px;
}

.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1),
.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):link,
.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):hover,
.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):active,
.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

<p
    className="c0"
  >
    A video of a talk I gave
  </p>,
]
`;

exports[`RichText renders correctly with a linked item 1`] = `
Array [
  .c0 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #252A31;
  line-height: 20px;
  text-align: left;
  margin: 0;
  margin-bottom: 8px;
}

.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1),
.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):link,
.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):hover,
.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):active,
.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

<p
    className="c0"
  >
    An intro to a component.
  </p>,
  "
",
  .c0 {
  width: 32px;
  height: 32px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  fill: currentColor;
}

.c1 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #252A31;
  line-height: 20px;
  text-align: left;
  margin: 0;
}

.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1),
.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):link,
.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):hover,
.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):active,
.c1 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

.c2 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #5F738C;
  line-height: 20px;
  text-align: left;
  margin: 0;
}

.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1),
.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):link,
.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):hover,
.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):active,
.c2 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

<div
    css={
      Array [
        "
        display: flex;
        align-items: center;
      ",
      ]
    }
  >
    <svg
      className="c0"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 24 24"
    >
      <path
        d="M20.696 4.8H3.304C2.584 4.8 2 5.384 2 6.104V17.89a1.269 1.269 0 001.304 1.259h2.609v3.478a.435.435 0 00.683.357l5.54-3.835h8.56c.72 0 1.304-.584 1.304-1.304V6.103c0-.72-.584-1.304-1.304-1.304zm-2.609 7.341a1.304 1.304 0 11-2.609 0 1.304 1.304 0 012.609 0zm-12.174 0a1.304 1.304 0 112.609 0 1.304 1.304 0 01-2.609 0zm4.783 0a1.304 1.304 0 112.608 0 1.304 1.304 0 01-2.608 0z"
      />
    </svg>
    <blockquote>
      <p
        className="c1"
      >
        You never really understand a person until you consider things from [their] point of view â€” until you climb into [their] skin and walk around in it.
      </p>
      <footer>
        <p
          className="c2"
        >
          <p>
            Atticus Finch in 
            <cite>
              To Kill a Mockingbird 
            </cite>
            by Harper Lee
          </p>
        </p>
      </footer>
    </blockquote>
  </div>,
]
`;

exports[`RichText renders correctly with a list with links 1`] = `
.c3 {
  width: 24px;
  height: 24px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  fill: currentColor;
}

.c5 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-weight: 500;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
}

.c5,
.c5:link,
.c5:visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c5:hover,
.c5:active,
.c5:focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

.c1 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 4px;
}

.c1:last-child,
.c1:last-of-type {
  margin: 0;
}

.c1 .Text__StyledText-sc-19qtt4y-0 {
  line-height: inherit;
  font-size: inherit;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 8px 0 0;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 20px;
}

.c2 .CarrierLogo__StyledCarrierLogo-d4doc9-1 {
  height: 16px;
  width: 16px;
}

.c2 .CarrierLogo__StyledCarrierLogo-d4doc9-1 img {
  height: 16px;
  width: 16px;
}

.c2 svg {
  height: 16px;
  width: 16px;
}

.c4 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  line-height: 20px;
  color: #252A31;
  list-style: none;
  padding: 0;
  margin: 0;
  margin-bottom: 8px;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <div
      className="c2"
      size="normal"
      type="primary"
    >
      <svg
        className="c3"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5z"
        />
      </svg>
    </div>
    <span
      className="c4"
    >
      A video of a talk I gave at the 
      <a
        className="c5"
        href="http://www.writethedocs.org/conf/prague/2019/"
        onClick={[Function]}
        rel=""
      >
        Write the Docs Prague 2019
      </a>
       conference.
    </span>
  </li>
  <li
    className="c1"
  >
    <div
      className="c2"
      size="normal"
      type="primary"
    >
      <svg
        className="c3"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5z"
        />
      </svg>
    </div>
    <span
      className="c4"
    >
       Or read a text version of the talk.
    </span>
  </li>
</ul>
`;

exports[`RichText renders correctly with a simple list 1`] = `
.c3 {
  width: 24px;
  height: 24px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  fill: currentColor;
}

.c1 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-bottom: 4px;
}

.c1:last-child,
.c1:last-of-type {
  margin: 0;
}

.c1 .Text__StyledText-sc-19qtt4y-0 {
  line-height: inherit;
  font-size: inherit;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 8px 0 0;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 20px;
}

.c2 .CarrierLogo__StyledCarrierLogo-d4doc9-1 {
  height: 16px;
  width: 16px;
}

.c2 .CarrierLogo__StyledCarrierLogo-d4doc9-1 img {
  height: 16px;
  width: 16px;
}

.c2 svg {
  height: 16px;
  width: 16px;
}

.c4 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  line-height: 20px;
  color: #252A31;
  list-style: none;
  padding: 0;
  margin: 0;
  margin-bottom: 8px;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <div
      className="c2"
      size="normal"
      type="primary"
    >
      <svg
        className="c3"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5z"
        />
      </svg>
    </div>
    <span
      className="c4"
    >
      A video of a talk I gave at the Write the Docs Prague 2019 conference.
    </span>
  </li>
  <li
    className="c1"
  >
    <div
      className="c2"
      size="normal"
      type="primary"
    >
      <svg
        className="c3"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5z"
        />
      </svg>
    </div>
    <span
      className="c4"
    >
       Or read a text version of the talk.
    </span>
  </li>
</ul>
`;

exports[`RichText renders correctly with an image 1`] = `
Array [
  .c0 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #252A31;
  line-height: 20px;
  text-align: left;
  margin: 0;
  margin-bottom: 8px;
}

.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1),
.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):link,
.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):hover,
.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):active,
.c0 a:not(.TextLink__StyledTextLink-sc-1bvlje4-1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

<p
    className="c0"
  >
    An awesome intro
  </p>,
  <picture>
    <source
      media="(min-width: 1500px)"
      srcSet="https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/18b254b8-ee42-42ca-9c5a-9105fa9a473d/Slide6.PNG?w=1200&fit=scale"
    />
    <source
      media="(min-width: 720px)"
      srcSet="https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/18b254b8-ee42-42ca-9c5a-9105fa9a473d/Slide6.PNG?w=1100&fit=scale"
    />
    <source
      media="(min-width: 400px)"
      srcSet="https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/18b254b8-ee42-42ca-9c5a-9105fa9a473d/Slide6.PNG?w=500&fit=scale"
    />
    <img
      alt="A map of an experience of a plane trip from London To New York."
      css={
        Array [
          "
          max-width: 100%;
        ",
        ]
      }
      src="https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/18b254b8-ee42-42ca-9c5a-9105fa9a473d/Slide6.PNG?w=300&fit=scale"
    />
  </picture>,
]
`;

exports[`RichText renders correctly with links 1`] = `
.c2 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-weight: 500;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: color 0.15s ease-in-out;
  transition: color 0.15s ease-in-out;
}

.c2,
.c2:link,
.c2:visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c2:hover,
.c2:active,
.c2:focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

.c0 {
  font-family: 'Roboto',-apple-system,'.SFNSText-Regular','San Francisco','Segoe UI','Helvetica Neue','Lucida Grande',sans-serif;
  font-size: 14px;
  font-weight: 400;
  color: #252A31;
  line-height: 20px;
  text-align: left;
  margin: 0;
  margin-bottom: 8px;
}

.c0 a:not(.c1),
.c0 a:not(.c1):link,
.c0 a:not(.c1):visited {
  color: #007F6D;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: 500;
}

.c0 a:not(.c1):hover,
.c0 a:not(.c1):active,
.c0 a:not(.c1):focus {
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #009882;
}

<p
  className="c0"
>
  Read a 
  <a
    className="c1 c2"
    href="/articles/seeing-your-docs-through-different-eyes"
    onClick={[Function]}
    rel=""
  >
    text version of the talk
  </a>
   I gave at the 
  <a
    className="c1 c2"
    href="http://www.writethedocs.org/conf/prague/2019/"
    onClick={[Function]}
    rel=""
  >
    Write the Docs Prague 2019
  </a>
   conference.
</p>
`;
